### PR TITLE
Fix #36: Implement support for invalid SSL certificate on Web Flow related services

### LINKED ISSUE #36: Implement support for invalid SSL certificate on Web Flow related services
We currently expect communication with either valid SSL certificate or over HTTP.

Web Flow application should support `powerauth.service.ssl.acceptInvalidSslCertificate` attribute for communication with PowerAuth 2.0 Server and similar attributes for other application it communicates with.

### PR DESCRIPTION
**Pull request by romanstrobl**
Added trusting of all certificates during initializations of various clients

### CODE REVIEW

File: powerauth-webauth-authentication/src/main/java/io/getlime/security/powerauth/lib/webauth/authentication/configuration/WebAuthServicesConfiguration.java [hunk 1]
```diff
@@ -57,7 +81,44 @@ public CredentialStoreClient defaultCredentialStoreClient() {
...
+        if (acceptInvalidSslCertificateInNextStep) {
+            trustAllCertificates();
+        }
+        return client;
     }
 
+    /**
+     * Activate trust in all SSL certificates including invalid ones for non-production use.
+     */
+    private void trustAllCertificates() {
```
Comment (petrdvorak) [inline powerauth-webauth-authentication/src/main/java/io/getlime/security/powerauth/lib/webauth/authentication/configuration/WebAuthServicesConfiguration.java]: This method disables SSL certificate handling everywhere - not just on one or another interface. We should either:

- implement SSL disabling in such way that it works per client (Next Step vs. Credential Store), or...
- have only one property "acceptInvalidSslCertificate" that disables SSL check for all services.

File: powerauth-webauth-client/src/main/java/io/getlime/security/powerauth/app/webauth/demo/configuration/WebAuthServiceConfiguration.java [hunk 2]
```diff
@@ -73,7 +86,45 @@ public String getClientSecret() {
...
+            trustAllCertificates();
+        }
+        return client;
+    }
+
+    /**
+     * Activate trust in all SSL certificates including invalid ones for non-production use.
+     */
+    private void trustAllCertificates() {
+        HttpsURLConnection.setDefaultHostnameVerifier((hostname, session) -> true);
```
Comment (petrdvorak) [inline powerauth-webauth-client/src/main/java/io/getlime/security/powerauth/app/webauth/demo/configuration/WebAuthServiceConfiguration.java]: Same issue as above - this setting works for all `HttpsURLConnection` instances, there is no benefit in having two separate attributes for it and current behavior is surprising ("I only disabled SSL on Next Step, and it's disabled on Credential Store as well), hence incorrect.

File: powerauth-webauth/src/main/java/io/getlime/security/powerauth/app/webauth/configuration/PowerAuthWebServiceConfiguration.java [hunk 3]
```diff
@@ -60,14 +70,55 @@ public PowerAuthServiceClient powerAuthClient(Jaxb2Marshaller marshaller) {
...
+        if (acceptInvalidSslCertificate) {
+            trustAllCertificates();
+        }
         return client;
     }
 
+    /**
+     * Activate trust in all SSL certificates including invalid ones for non-production use.
+     */
+    private void trustAllCertificates() {
```
Comment (petrdvorak) [inline powerauth-webauth/src/main/java/io/getlime/security/powerauth/app/webauth/configuration/PowerAuthWebServiceConfiguration.java]: Why do we disable SSL in multiple Maven modules in a single project?

File: powerauth-webauth-authentication/src/main/java/io/getlime/security/powerauth/lib/webauth/authentication/configuration/WebAuthServicesConfiguration.java (hunk 1 above)
Comment (romanstrobl) [inline powerauth-webauth-authentication/src/main/java/io/getlime/security/powerauth/lib/webauth/authentication/configuration/WebAuthServicesConfiguration.java]: You are right. I think it is enough to have one property, this setting is used for development as far as I understand.

File: powerauth-webauth/src/main/java/io/getlime/security/powerauth/app/webauth/configuration/PowerAuthWebServiceConfiguration.java (hunk 3 above)
Comment (romanstrobl) [inline powerauth-webauth/src/main/java/io/getlime/security/powerauth/app/webauth/configuration/PowerAuthWebServiceConfiguration.java]: It is not the same project - one commit affects powerauth-webauth.war and second commit affects powerauth-webauth-client.war. I thought we needed to disable SSL certificate check in each war file separately? If not, then I will change it.

Comment (romanstrobl) [inline powerauth-webauth/src/main/java/io/getlime/security/powerauth/app/webauth/configuration/PowerAuthWebServiceConfiguration.java]: Ah, I see now, you probably mean duplicate setting in powerauth-webauth and powerauth-webauth-authentication projects. Yes, that should be fixed.

File: powerauth-webauth-client/src/main/java/io/getlime/security/powerauth/app/webauth/demo/configuration/WebAuthServiceConfiguration.java (hunk 2 above)
Comment (romanstrobl) [inline powerauth-webauth-client/src/main/java/io/getlime/security/powerauth/app/webauth/demo/configuration/WebAuthServiceConfiguration.java]: Fixed, same configuration property is used in all projects.

File: powerauth-webauth-authentication/src/main/java/io/getlime/security/powerauth/lib/webauth/authentication/configuration/WebAuthServicesConfiguration.java [hunk 4]
```diff
@@ -43,12 +47,28 @@
...
+    /**
      * Default credential store client.
      * @return Credential store client.
      */
     @Bean
     public CredentialStoreClient defaultCredentialStoreClient() {
-        return new CredentialStoreClient(credentialServerServiceUrl);
+        CredentialStoreClient client = new CredentialStoreClient(credentialServerServiceUrl);
+        // whether invalid SSL certificates should be accepted
+        if (acceptInvalidSslCertificate) {
```
Comment (petrdvorak) [inline powerauth-webauth-authentication/src/main/java/io/getlime/security/powerauth/lib/webauth/authentication/configuration/WebAuthServicesConfiguration.java]: We are not disabling SSL validation for the client, but for all server application...

**Pull request merged**